lib/deep-omit/index.ts

```diff
@@ -1,67 +1,46 @@
 import { AnyRecord } from "../any-record";
 import { Builtin } from "../built-in";
-import { DeepModify } from "../deep-modify";
 
-export type DeepOmit<Type, Filter extends DeepModify<Type>> = Type extends Builtin
+export type DeepOmit<Type, Filter> = Type extends Builtin
   ? Type
   : Type extends Map<infer Keys, infer Values>
   ? Filter extends Map<Keys, infer FilterValues>
-    ? FilterValues extends DeepModify<Values>
```

Comment:
9 conditional types were removed in `DeepOmit` and `DeepPick` which improves performance of these utility types after this change